bubblewrap.c

```diff
@@ -45,13 +45,29 @@
 static uid_t uid;
 static gid_t gid;
 static bool is_privileged;
+static bool did_setuid;
 static const char *argv0;
 static const char *host_tty_dev;
 static int proc_fd = -1;
 static char *opt_exec_label = NULL;
 static char *opt_file_label = NULL;
 static bool no_reaper;
 
+
+/* by default enabled caps for privileged user.  */
+#define BASE_CAPS0 (CAP_TO_MASK (CAP_NET_RAW) | CAP_TO_MASK (CAP_NET_BIND_SERVICE) | CAP_TO_MASK (CAP_AUDIT_WRITE) \
+                    | CAP_TO_MASK (CAP_DAC_OVERRIDE) | CAP_TO_MASK(CAP_SETFCAP) | CAP_TO_MASK(CAP_SETPCAP) \
+                    | CAP_TO_MASK (CAP_SETGID) | CAP_TO_MASK (CAP_SETUID) | CAP_TO_MASK (CAP_MKNOD) | CAP_TO_MASK (CAP_CHOWN) \
+                    | CAP_TO_MASK (CAP_FOWNER) | CAP_TO_MASK (CAP_FSETID) | CAP_TO_MASK (CAP_KILL) \
+                    | CAP_TO_MASK (CAP_SYS_CHROOT))
+#define BASE_CAPS1 (CAP_TO_MASK (CAP_AUDIT_READ))
+
+static uint32_t requested_caps[2] = {BASE_CAPS0, BASE_CAPS1};
+
+/* low 32bit caps needed */
+#define REQUIRED_SETUP_CAPS_0 (BASE_CAPS0 | CAP_TO_MASK (CAP_NET_ADMIN) | CAP_TO_MASK (CAP_SYS_ADMIN))
```

Comment:
You dropped CHROOT, SETUID and SETGID from REQUIRED_CAPS. These are needed for chroot() and user namespaces to work in the setup.